apps/prairielearn/elements/pl-big-o-input/pl-big-o-input.mustache

```diff
@@ -17,28 +17,56 @@
         {{#inline}}<span class="form-inline d-inline-block ml-2">{{/inline}}
             <span id="big-o-input-{{uuid}}" class="input-group big-o-input">
                 <span class="input-group-prepend">
-                    <span class="input-group-text">${{type}}($</span>
+                    <span class="input-group-text" id="pl-big-o-input-{{uuid}}-label">${{type}}($</span>
                 </span>
-                <input name="{{name}}" type="text" autocomplete="off" class="form-control big-o-input-input {{^show_info}}rounded-right{{/show_info}}" size="{{size}}" {{^editable}}disabled{{/editable}} {{#raw_submitted_answer}}value="{{raw_submitted_answer}}"{{/raw_submitted_answer}} placeholder="{{placeholder}}"/>
+                <input
+                    name="{{name}}"
+                    type="text"
+                    autocomplete="off"
+                    class="form-control big-o-input-input"
+                    size="{{size}}"
+                    {{^editable}}disabled{{/editable}}
+                    {{#raw_submitted_answer}}value="{{raw_submitted_answer}}"{{/raw_submitted_answer}}
+                    placeholder="{{placeholder}}"
+                    aria-describedby="pl-big-o-input-{{uuid}}-label pl-integer-input-{{uuid}}-suffix"
+                />
                 <span class="input-group-append">
-                    <span class="input-group-text">$)$</span>
+                    <span
+                        class="input-group-text"
+                        id="pl-big-o-input-{{uuid}}-suffix">$)$</span>
```

Comment:
What is a screenreader going to do if an input is described by the strings `O(` and `)`, respectively? Is this tested?

Reply:
I haven't tested this out (copying the other attributes on this), but will take a look when I get the chance. Haven't done any real testing with screen readers before, so will ask in DM / in the next dev meeting.

Reply:
Note from meeting: Test from screenreader and see what gets announced. If it does something weird, use an `arialabel` to get something more useful.

Reply:
Tested with this screen reader extension: https://addons.mozilla.org/en-US/firefox/addon/read-aloud/

Without making any changes, the reader was surprisingly natural! It ignored parenthesis and just read "O(" as "Oh". Had no issue with the omegas and thetas either. I think the performance is sufficient that we don't need to make any changes.